docs/source/enterprise-guide/change-management.mdx

```diff
@@ -0,0 +1,468 @@
+
+Rolling back the changes is simply a matter of flipping traffic to the "green"
+variant.
+
+## Changes to a data source
+
+Changing a data source safely requires three separate releases to avoid
+downtime. This process should be very familiar; it is standard practice for any
+kind of database migration or API change in a service-oriented architecture.
+
+1. Release #1: Make a backwards-compatible change.
+
+   For example, add a column to a database and backfill the data using an async
+   process.
+
+2. Release #2: Update calling sites.
+
+   For example, change SQL queries to use new column.
+
+3. Remove unneeded code/data.
+
+   For example, drop the unused column.
+
+### Progressive delivery techniques
+
+**Feature flags**: we may choose to leverage feature flags in Release #2,
+enabling a quick rollback if we discover a problem in the data or
+implementation. In order to "pin" users to a specific flag value and ensure a
+consistent experience during the rollout, pass a user-identifying header though
+each layer of the system.
+
+We can configure the gateway to forward headers to subgraph services by
+[customizing the subgraph data source][customizereqs].
+
+[customizereqs]:
+  https://www.apollographql.com/docs/federation/gateway/#customizing-requests
+
+## Changes to subgraph resolvers
+
+Changing the behavior of a resolver (such as calling a new data source, changing
+the computation of a value, or optimizing execution by adding
+[Dataloaders](dataloader)) may be performed in a single release for low-risk
+changes.
+
+Because a resolver may be called in various contexts — nested at different
+layers of an operation — it can be challenging to reason about the effects of
+our change. We may choose to progressively release the change using feature
+flags or canary deploys.
+
+[dataloader]:
+  https://www.apollographql.com/docs/apollo-server/data/data-sources/#using-with-dataloader
+
+### Progressive delivery techniques
+
+**Feature flags**: we may choose to leverage feature flags, breaking up the
+release into multiple steps, enabling a quick rollback if we discover a problem
+in the implementation. In order to "pin" users to a specific flag value and
+ensure a consistent experience during the rollout, pass a user-identifying
+header though each layer of the system.
+
+1. Author the resolver to provide both old and new codepaths gated by a feature
+   flag.
+2. Release the new code. The new codepath is enabled for zero users by default,
+   so the release should have no impact on current behavior.
+3. Progressively enable the feature flag for more users, monitoring error rates
+   or support inquiries.
+4. Once the feature flag is enabled for 100% of users, remove the old codepath
+   and perform a clean-up release.
+
+**Canary deploys**: instead of adding complexity to our resolver by supporting
+two codepaths and selecting one based on a feature flag, we may choose to simply
```

Comment:
As mentioned with "may" to "might" before, if this "may" is more accurately "can", replace with "can" instead